sherpa/astro/data.py

```diff
@@ -2182,6 +2182,18 @@ def get_noticed_channels(self):
         chans = self.channel
         mask = self.get_mask()
         if mask is not None:
+
+            # This is added to address issue #361
+            #
+            # If there is a quality filter then the mask may be
+            # smaller than the chans array. It is not clear if this
+            # is the best location for this. If it is, then are there
+            # other locations where this logic is needed?
+            #
+            if self.quality_filter is not None and \
+               self.quality_filter.size != mask.size:
+                chans = chans[self.quality_filter]
+
             chans = chans[mask]
```

Comment:
Unless I am missing something, the code in lines 2193-2195 are a no-op since line 2197 will override the value of chans.

Reply:
While I haven't looked at this for a while, I think what is important hers is that `chans` and `mask` have the same size. Without this addition there are times when this doesn't hold - i.e. we need to subset `chans` to match the size of `mask`.

Does this make sense?

Reply:
The issue I have with this PR is that it's not obvious to me if this is just a band aid, hiding a more serious problem, of if this is the best place to place this fix. However, at some point a band aid is better than nothing.

Reply:
So for the code segment, are we certain that the quality_flag True elements is always equal to the mask size?      
     (np.where(self.quality_filter == True))[0].size == mask,size

Reply:
All I know for sure is that there is a case (#361) where the fix above stopped the warning. I was hoping that this PR would lead to someone who did understand the filtering/grouping internals would be able to say "yes" or "this should be done elsewhere", but that hasn't happened. So now I think we either take this under the assumption that it doesn't seem to make things worse and does fix a known issue, or decide that it's too risky and we should wait until someone can fix this properly.